### DISCUSSION TITLE
Forwarding Alerts

### DISCUSSION THREAD

**Discussion by kiaraho** [Security-Onion-Solutions/securityonion#8295] [Unsupported Versions] [answered]
I'm trying to forward alerts using this [original event forwarding](https://docs.securityonion.net/en/latest/logstash.html#forwarding-events-to-an-external-destination) template in a distributed environment. Using `if [dataset] =~ "alert"` in my output results in zero alerts being forwarded to my external receiver. However, if I switch to `if [module] =~ "playbook" or [module] =~ "suricata"` the alerts start coming in, but leaves out Ossec alerts as expected.

Also, `if [module] =~ "ossec"  and [dataset] =~ "alert"` on its own does not work either. What I'm I missing?

**Comment by btdewey** [comment, marked as answer]
Did you copy the pipeline from the default location and then modify it?  I think you need to put each forwarder statement in the proper logstash pipeline file.  I'm not at my system so that path may not be exactly right, but should get you in the right place.  Look for a .jinja file that has the name of the log source you want to forward. 

 /opt/so/saltstack/**default**/salt/logstash/pipelines/config/

**Reply by kiaraho** [reply]
I followed @weslambert answer in #3456 and created the following jinja file in the `/opt/so/saltstack/local/salt/logstash/pipelines/config/custom/` directory.
```
filter {
  if [module] =~ "playbook" or [module] =~ "suricata"  or [dataset] =~ "alert" {
    clone {
        clones => ["so-alert-clone"]
        add_tag => ["alert-clone"]
    }
  }
}
output {
  if "alert-clone" in [tags] {
    udp {
      host => "192.168.x.x"
      port => 10011
      codec => "json_lines"
    }
  }
}
```
As far as I can tell, only Suricata alerts are being forwarded to the receiver at `192.168.x.x`. No Wazuh alerts are making it through.

**Reply by btdewey** [reply]
Can you try this?  I think you need to filter on the event.dataset field not the dataset field.

```
filter {
  if [event][dataset] =~ "alert" {
    clone {
        clones => ["so-alert-clone"]
        add_tag => ["alert-clone"]
    }
  }
}
output {
  if "alert-clone" in [tags] {
    udp {
      host => "192.168.x.x"
      port => 10011
      codec => "json_lines"
    }
  }
}
```
